### ISSUE TITLE
Problems running on OSX (12.6)

### ISSUE THREAD

**Issue by bernii** [WasmEdge/wasmedge_hyper_demo#8]
Trying to run an example on my machine but I'm running into some weird issues, wondering if there's an easy explanation here

```
$ cargo --version                                                                                                                                                                                           
cargo 1.65.0 (4bc8f24d3 2022-10-20)
$ cargo build --target wasm32-wasi && wasmedge target/wasm32-wasi/debug/lsat-proxy.wasm                                                                                                                     
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
2022-11-22T15:32:18.202591Z  INFO lsat_proxy: Listening on http://0.0.0.0:8080
// when trying to curl from the second console
Error: Os { code: 52, kind: Unsupported, message: "Function not implemented" }
```

Looking at the code it errors exactly here when attempting to accept the incoming connection
```
 let (stream, _) = listener.accept().await?;
```

Not sure where to dig from here.

Edit: side note: running on `M1` mac

**Comment by juntao** [comment]
Yeah. We have a problem with WASI implementation for Mac. You can run it on Linux or Docker for Mac OSX. But to run it on Mac, we need this PR to merge:

https://github.com/WasmEdge/WasmEdge/issues/2104

**Comment by bernii** [comment]
thanks for the info! 

just FYI the problematic line is this one https://github.com/second-state/wasmedge_wasi_socket/blob/bcf4fe2f19ac4629d1d56ff0f4ac882a5babdfa3/src/socket.rs#L656

I see my investigation matches the PR you've mentioned 👍

**Comment by MediosZ** [comment]
This should be fixed since [this pr](https://github.com/WasmEdge/WasmEdge/issues/2104) has been merged.

**Comment by ekzeb** [comment]
the same issue on 2,9 GHz 6-Core Intel Core i9 Ventura
wasmedge version 0.11.2

**Comment by bromaniac** [comment]
Same problem here.

2,4 GHz Quad-Core Intel Core i5 MPB 13" 2019
Ventura 13.2.1 (22D68)
wasmedge version 0.11.2

**Comment by MediosZ** [comment]
I tried this example `https://github.com/second-state/wasmedge_wasi_socket/tree/main/examples/http_server`, and it worked on Intel MacBook, 12.6.3(Monterey), with wasmedge 0.12.1.
I guess there's some breaking change in macOS 13(Ventura). And as https://github.com/WasmEdge/WasmEdge/issues/2056 mentioned, it's not working on M1 MacBooks with 13.2.1 as well.

**Comment by juntao** [comment]
Hmm, what's the error message on Mac OSX 13?

**Comment by MediosZ** [comment]
But the GitHub runner (macos 13) passes the test, check out [this ci log](https://github.com/MediosZ/wasmedge_wasi_socket/actions/runs/5010169694/jobs/8979797752).

**Comment by juntao** [comment]
The GitHub actions CI runner uses Intel based macs.

**Comment by MediosZ** [comment]
I believe the error occurs because the pr that adds networking functions on macOS is merged but not shipped with version 0.11.2.
And now this error should not occur anymore in version 0.12.1.
The test log can be found [here](https://github.com/MediosZ/wasmedge_wasi_socket/tree/test-mac).
